src/mobi/hsz/idea/gitignore/indexing/IgnoreEntryOccurrence.java

```diff
@@ -191,18 +167,14 @@ public static synchronized IgnoreEntryOccurrence deserialize(@NotNull DataInput
             return null;
         }
 
-        final IgnoreEntryOccurrence entry = new IgnoreEntryOccurrence(url);
-        int size = in.readInt();
-
-        @SuppressWarnings("unchecked")
-        Pair<Pattern, Boolean>[] items = (Pair<Pattern, Boolean>[]) new Pair[size];
+        final int size = in.readInt();
+        final ArrayList<Pair<Pattern, Boolean>> items = ContainerUtil.newArrayList();
         for (int i = 0; i < size; i++) {
             final Pattern pattern = Pattern.compile(in.readUTF());
```

Comment:
Btw, Pattern.compile is quite slow, I think that eventually it would be better to compile the pattern somewhere near matching (with the cache of String -> Pattern)

Reply:
Good catch. I've modified it to store in index regex Strings only.
Those strings will be converted to Pattern objects during the runtime (when checking rules against files) - and cached.
This approach is much better because Index can store some .npmignore, .eslintignore rules, that may not be used  for Project Tree testing - so Patterns have been created without reason.
Performance kick (lower RAM consumption) expected. ;)

Reply:
e3a4e2845deeb31d88549f1e6c35b8d51f2fde8d